### PR TITLE
feat: drop support for PHP < 8.1

### PR DESCRIPTION
**Pull request by Chris53897**
https://github.com/TheBigBrainsCompany/TbbcMoneyBundle/issues/153

drop PHPUnit 9 support.
replace php-http/guzzle6-adapter with php-http/guzzle7-adapter
bump dev dependencies
Migrate phpUnit config to version 10.
Improve some Tests

### CODE REVIEW

**Comment by Chris53897** [comment]
@johanwilfer PR is ready for review. The Psalm errors should be handled in a seperate PR.

File: composer.json [hunk 1]
```diff
@@ -39,14 +39,14 @@
     "require-dev": {
         "ext-sqlite3": "*",
         "symfony/browser-kit": "^5.4|^6.0|^7.0",
-        "doctrine/doctrine-bundle": "^1.12|^2.0",
```
Comment (johanwilfer) [inline composer.json:42]: One request here, can you check so we don't bump this to the latest version but instead try to use the minimal version that would work (the opposite of how you would do for an application using this) - to allow people an easier upgrade path?

Comment (Chris53897) [reply composer.json:42]: This is in the require-dev section. So i guess this should not make the upgrade path harder. A new major version for this PR is planned anyway.

Comment (Chris53897) [reply composer.json:42]: https://packagist.org/packages/doctrine/doctrine-bundle#2.11.0 adds the support for symfony 7

**Comment by johanwilfer** [comment]
@Chris53897 Can you update https://github.com/TheBigBrainsCompany/TbbcMoneyBundle/blob/master/change-log-6.0-and-versions.md in this branch as well then and this is also good to be merge to master (for the upcomming 6.x version)?

File: composer.json (hunk 1 above)
Comment (johanwilfer) [reply composer.json:41]: Ah, true - thanks!

**Comment by Chris53897** [comment]
I have upgraded it. this can be finetuned before first tag of 6.x

**Comment by johanwilfer** [comment]
Great!

**Pull request merged**